test/routes/article-list.test.ts

```diff
@@ -3,23 +3,65 @@ import { Next, Response } from 'koa';
 import articleList from '../../src/routes/article-list';
 import createContext from '../context';
 import runMiddleware from '../middleware';
+import InMemoryArticles from '../../src/adaptors/in-memory-articles';
 
-const makeRequest = async (next?: Next): Promise<Response> => {
+const makeRequestEmptyLst = async (next?: Next): Promise<Response> => {
   const context = createContext();
+  const articles = new InMemoryArticles();
 
-  return runMiddleware(articleList(context.router), context, next);
+  return runMiddleware(articleList(articles, context.router), context, next);
+};
+
+const makeRequestList = async (next?: Next): Promise<Response> => {
+  const context = createContext();
+  const articles = new InMemoryArticles();
+
+  (async (): Promise<void> => {
+    await Promise.all([
+      articles.add({
+        '@type': 'http://schema.org/Article',
+        '@id': '/articles/09560',
+        'http://schema.org/name': 'Homo naledi, a new species of the genus Homo from the Dinaledi Chamber, South Africa',
+      }),
+      articles.add({
+        '@type': 'http://schema.org/Article',
+        '@id': '/articles/24231',
+        'http://schema.org/name': 'The age of Homo naledi and associated sediments in the Rising Star Cave, South Africa',
+      }),
+    ]);
+  })();
+
+  return runMiddleware(articleList(articles, context.router), context, next);
 };
```

Comment:
Rather than have two, I'd make `articles` an optional argument (default to an empty `InMemoryArticles`), then add in the items in the test itself.